### PR TITLE
add validate with regex

### PR DESCRIPTION
**Pull request by GuilhermeSanches**
I think the best practice is to add regex validation in the input, so users can not enter float numbers or characters.

### CODE REVIEW

File: test/FactorialServiceSpec.js [hunk 1]
```diff
@@ -15,10 +15,18 @@ describe('FactorialService.js', function() {
     
     it('should null if an invalid param was given', function() {
         expect(FactorialService.calculate(-1)).toBe(null);
-        expect(FactorialService.calculate("1")).toBe(null);
```
Comment (edysegura) [inline test/FactorialServiceSpec.js:18]: Hi @GuilhermeSanches! 

Why did you remove this test case? It is necessary to make sure the method calculate will not support a number inside into a string.

When we have tests case we refactor the code without change the test to make sure if the business logic still valid with the new code.

File: app/js/FactorialService.js [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
 
     var _results = [1, 1];
-
+      
+    var _reg = new RegExp('^[0-9]+$');   
+     
     function _isNumberValid(number) {
-        return (number !== undefined)
-            && (number > -1)
-            && (typeof(number) !== 'string');
```
Comment (edysegura) [inline app/js/FactorialService.js:8]: We need this validation to make sure we are not dealing with a number inside into a string

**Comment by edysegura** [comment]
Hi @GuilhermeSanches!

The regex for validation is a great idea but the test case should not be removed when refactoring the code. It is against TDD.

Thank you to colaborate.

File: test/FactorialServiceSpec.js (hunk 1 above)
Comment (GuilhermeSanches) [inline test/FactorialServiceSpec.js:18]: @edysegura the input used is of type text, then in my opinion should accept instead a string equal to "5", for example. Since it can not accept string, declaring the type number would not be more appropriate? thank you for attention. :)

Comment (edysegura) [inline test/FactorialServiceSpec.js:18]: The controller should be one responsible to parse the input text and send
it to the service.

Look at FactorialController.js

Comment (GuilhermeSanches) [inline test/FactorialServiceSpec.js:18]: Yeah, but this caused a problem with number float. How is done parse to int, if to write 4.5, the code sends only 4, generating result 24... which theoretically should not occur. Anyway. Thanks for the answers.

Comment (edysegura) [inline test/FactorialServiceSpec.js:18]: Please take a look about the rules to calculate the factorial! It only
apply for positive and natural number.

https://pt.m.wikipedia.org/wiki/Fatorial
